test/routes/token.js

```diff
@@ -25,8 +26,12 @@ function joiNotAllowed(err, param) {
 
 describe('/token POST', function () {
   // route validation function
-  function v(req, cb) {
-    route.validate.payload(req, {}, cb);
+  function v(req, ctx, cb) {
+    if (typeof ctx === 'function' && ! cb) {
+      cb = ctx;
+      ctx = undefined;
+    }
+    Joi.validate(req, route.validate.payload, {context: ctx}, cb);
```

Comment:
This copies what hapi does to pass around context internally, but I made sure to also add some live API tests covering the `$headers.authorization` reference stuff, in case it breaks in future.